test/gpu/fuse_hip_gemm.cpp

```diff
@@ -0,0 +1,102 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <basic_ops.hpp>
+#include <migraphx/auto_contiguous.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/gpu/compile_hipblaslt.hpp>
+#include <migraphx/gpu/fuse_ops.hpp>
+#include <migraphx/gpu/hip_gemm.hpp>
+#include <migraphx/gpu/lowering.hpp>
+#include <migraphx/gpu/target.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/op/allocate.hpp>
+#include <migraphx/op/common.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/program.hpp>
+#include <pointwise.hpp>
+#include <test.hpp>
+
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_ENABLE_HIPBLASLT_GEMM)
+
+void run_lowering(migraphx::program& p, bool offload_copy = false)
+{
+    auto ctx = migraphx::gpu::context{};
+    migraphx::run_passes(
+        *p.get_main_module(),
+        {migraphx::auto_contiguous{}, migraphx::gpu::lowering{&ctx, offload_copy}});
+}
+
+void run_fuse_ops(migraphx::program& p)
+{
+    migraphx::run_passes(p, {migraphx::gpu::fuse_ops{}, migraphx::dead_code_elimination{}});
+}
+
+#if MIGRAPHX_USE_HIPBLASLT
+TEST_CASE(hip_gemm_pointwise_add)
+{
+    migraphx::shape s{migraphx::shape::float_type, {1, 3, 3}};
+    migraphx::program p1;
+    {
+        auto* mm = p1.get_main_module();
+        auto a   = mm->add_parameter("a", s);
+        auto b   = mm->add_parameter("b", s);
+        auto x   = mm->add_parameter("x", s);
+        auto dot = mm->add_instruction(migraphx::make_op("dot"), a, b);
+        auto add = add_pointwise(p1, "main:pointwise0", {dot, x}, single_pointwise("add"));
+        mm->add_return({add});
+    }
+    run_lowering(p1);
+    run_fuse_ops(p1);
+
+    migraphx::program p2;
+    {
+        auto* mm = p2.get_main_module();
+        auto a   = mm->add_parameter("a", s);
+        auto b   = mm->add_parameter("b", s);
+        auto x   = mm->add_parameter("x", s);
+
+        auto output = mm->add_instruction(migraphx::op::allocate{s, std::nullopt});
+
+        migraphx::op::dot dot_instance;
+        migraphx::gpu::hipblaslt_op hipblaslt_operator;
+        hipblaslt_operator.op = migraphx::gpu::hip_gemm<migraphx::op::dot>{dot_instance, 1, 1};
+        auto add              = mm->add_instruction(hipblaslt_operator, a, b, x, output);
+
+        mm->add_return({add});
+    }
+    EXPECT(p1.sort() == p2.sort());
+}
+#endif
+
+int main(int argc, const char* argv[])
+{
+#if MIGRAPHX_USE_HIPBLASLT
+    if(migraphx::enabled(MIGRAPHX_ENABLE_HIPBLASLT_GEMM{}) and migraphx::gpu::hipblaslt_supported())
+#endif
+        test::run(argc, argv);
+    return 0;
```

Comment:
Why you do have a preprocessor condition on MIGRAPHX_USE_HIPBLASLT here? It looks like it would run this test even if MIGRAPHX_USE_HIPBLASLT were not set?

Reply:
Made suggested changes.

Reply:
Seems this change has caused Windows build failure. I have reverted the change to confirm.

Reply:
Conditional removed as a result of modifying the test case to be run for both rocblas and hipblaslt.